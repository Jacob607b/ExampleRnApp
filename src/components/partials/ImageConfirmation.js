import React from "react";
import {
  View,
  Text,
  Image,
  StyleSheet,
  Modal,
  TouchableOpacity,
} from "react-native";
import { Button } from "react-native-paper";

export default function ImageConfirmation({ image, deleteFunction }) {
  const [visible, setVisible] = React.useState(true);

  return (
    <View>
    <Modal
      visible={visible}
      animationType="slide"
      onRequestClose={() => setVisible(false)}
      testID="modal"
    >
      <View
        style={{
          flex: 1,
          justifyContent: "center",
          alignItems: "center",
          fontWeight: "bold",
          fontSize: 20,
        }}
      >
        <Text style={{ fontSize: 20 }}>Does this look okay?</Text>

        <Image
          source={{ uri: image }}
          style={styles.image}
          resizeMode="center"
        />

        <TouchableOpacity onPress={() => setVisible(false)} testID="Yes">
          <Text style={{ fontSize: 20 }}>
            <Button mode="elevated" className="SuccessModal">
              Yes
            </Button>
          </Text>
        </TouchableOpacity>
        <TouchableOpacity onPress={() => deleteFunction(image)} testID="No">
          <Text style={{ fontSize: 20 }}>
            <Button mode="elevated" className="SuccessModal">
              No
            </Button>
          </Text>
        </TouchableOpacity>
      </View>
    </Modal>
    </View>
  );
}

const styles = StyleSheet.create({
  image: {
    width: "100%",
    height: "70%",
  },
  button: {
    marginTop: 10,
  },
  container: {
    backgroundColor: "white",
    padding: 20,
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "center",
    textAlign: "center",
    fontWeight: "bold",
    height: "70%",
    flex: 1,
  },
});
