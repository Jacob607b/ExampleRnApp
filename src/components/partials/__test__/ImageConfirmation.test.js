import React from "react";
import { fireEvent, render, screen } from "@testing-library/react-native";
import ImageConfirmation from "../ImageConfirmation";
describe("Image Confirmation", () => {
  test("Image Confirmation Buttons (Yes)", () => {
    const deleteFunction = jest.fn();

    render(<ImageConfirmation deleteFunction={deleteFunction} />);

    const yesButton = screen.getByTestId("Yes");
    expect(yesButton).toBeTruthy();
    fireEvent.press(yesButton);
  });
  test("Image Confirmation Buttons (No)", () => {
    const deleteFunction = jest.fn();

    render(<ImageConfirmation deleteFunction={deleteFunction} />);

    const noButton = screen.getByTestId("No");
    expect(noButton).toBeOnTheScreen();
    fireEvent.press(noButton);
  });

  test("Image Confirmation Snapshot", async () => {
    render(<ImageConfirmation />);
    expect(screen).toMatchSnapshot();
  });
  test("Modal close action", () => {
    const deleteFunction = jest.fn();

    render(<ImageConfirmation deleteFunction={deleteFunction} />);

    // Simulate modal close action
    const modal = screen.getByTestId("modal");
    fireEvent(modal, "onRequestClose");
  });
});
