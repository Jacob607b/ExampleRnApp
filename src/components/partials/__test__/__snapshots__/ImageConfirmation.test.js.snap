// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Confirmation Image Confirmation Snapshot 1`] = `
<View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    testID="modal"
    visible={true}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "fontSize": 20,
          "fontWeight": "bold",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        Does this look okay?
      </Text>
      <Image
        resizeMode="center"
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Object {
            "height": "70%",
            "width": "100%",
          }
        }
      />
      <View
        accessibilityState={
          Object {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          Object {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="Yes"
      >
        <Text
          style={
            Object {
              "fontSize": 20,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "backgroundColor": "rgb(247, 243, 249)",
                "borderRadius": 20,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 0,
                },
                "shadowOpacity": 0.15,
                "shadowRadius": 3,
              }
            }
            testID="button-container-outer-layer"
          >
            <View
              className="SuccessModal"
              collapsable={false}
              style={
                Object {
                  "backgroundColor": "rgb(247, 243, 249)",
                  "borderColor": "transparent",
                  "borderRadius": 20,
                  "borderStyle": "solid",
                  "borderWidth": 0,
                  "flex": undefined,
                  "minWidth": 64,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": 0,
                  },
                  "shadowOpacity": 0.3,
                  "shadowRadius": 1,
                }
              }
              testID="button-container"
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": true,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  Object {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 20,
                    },
                  ]
                }
                testID="button"
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    numberOfLines={1}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "textAlign": "left",
                        },
                        Object {
                          "color": "rgba(28, 27, 31, 1)",
                          "writingDirection": "ltr",
                        },
                        Array [
                          Object {
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          Array [
                            Object {
                              "marginHorizontal": 16,
                              "marginVertical": 9,
                              "textAlign": "center",
                            },
                            false,
                            Object {
                              "marginHorizontal": 24,
                              "marginVertical": 10,
                            },
                            undefined,
                            false,
                            Object {
                              "color": "rgba(103, 80, 164, 1)",
                              "fontFamily": "System",
                              "fontSize": 14,
                              "fontWeight": "500",
                              "letterSpacing": 0.1,
                              "lineHeight": 20,
                            },
                            undefined,
                          ],
                        ],
                      ]
                    }
                    testID="button-text"
                  >
                    Yes
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </Text>
      </View>
      <View
        accessibilityState={
          Object {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          Object {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="No"
      >
        <Text
          style={
            Object {
              "fontSize": 20,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "backgroundColor": "rgb(247, 243, 249)",
                "borderRadius": 20,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 0,
                },
                "shadowOpacity": 0.15,
                "shadowRadius": 3,
              }
            }
            testID="button-container-outer-layer"
          >
            <View
              className="SuccessModal"
              collapsable={false}
              style={
                Object {
                  "backgroundColor": "rgb(247, 243, 249)",
                  "borderColor": "transparent",
                  "borderRadius": 20,
                  "borderStyle": "solid",
                  "borderWidth": 0,
                  "flex": undefined,
                  "minWidth": 64,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": 0,
                  },
                  "shadowOpacity": 0.3,
                  "shadowRadius": 1,
                }
              }
              testID="button-container"
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": true,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  Object {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 20,
                    },
                  ]
                }
                testID="button"
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    numberOfLines={1}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "textAlign": "left",
                        },
                        Object {
                          "color": "rgba(28, 27, 31, 1)",
                          "writingDirection": "ltr",
                        },
                        Array [
                          Object {
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          Array [
                            Object {
                              "marginHorizontal": 16,
                              "marginVertical": 9,
                              "textAlign": "center",
                            },
                            false,
                            Object {
                              "marginHorizontal": 24,
                              "marginVertical": 10,
                            },
                            undefined,
                            false,
                            Object {
                              "color": "rgba(103, 80, 164, 1)",
                              "fontFamily": "System",
                              "fontSize": 14,
                              "fontWeight": "500",
                              "letterSpacing": 0.1,
                              "lineHeight": 20,
                            },
                            undefined,
                          ],
                        ],
                      ]
                    }
                    testID="button-text"
                  >
                    No
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </Text>
      </View>
    </View>
  </Modal>
</View>
`;
